enhancements/20240612-node-disk-manager-refactor.md

```diff
@@ -0,0 +1,67 @@
+# Node Disk Manager Refactor
+
+## Summary
+
+The Node Disk Manager is responsible for managing the raw device on the node. The current implementation is not flexible enough to support various storage backends. This enhancement is to refactor the Node Disk Manager to make it more flexible and extensible.
+
+### Related Issues
+
+https://github.com/harvester/harvester/issues/6014
+
+## Motivation
+
+We need to refactor the current implementation to make the Node Disk Manager support various storage backends. This could help define the storage provisioner easily.
+
+### Goals
+
+Refactor the Node Disk Manager to make it more flexible and extensible to support various storage backends. The refactor will decompose the individual storage provisioner.
```

Comment:
I would suggest removing the above sentence as it's pretty similar to the motivation section, and instead explicitly state the additional storage backends we wish to end up with, as goals, similar to what's in the related issue https://github.com/harvester/harvester/issues/6014. Maybe something like this?

- Add longhornv2 provisioner
- Add LVM provisioner
- Continue to support existing longhornv1 provisioning

Reply:
Thanks @tserong,
I rewrote this part to specify the Goals of this refactor.
Please help to recheck!